### DISCUSSION TITLE
Did I just lose my network??

### DISCUSSION THREAD

**Discussion by Jeppedy** [xyzroe/XZG#90] [Q&A] [answered]
The UZG-01 XZG UI popped up and told me there was a new Zigbee firmware available for my device.  I started the upgrade.  This led to that to another, and it seems the controller cannot talk to any of my devices.  I can't even allow new devices to join!
Did I just destroy my Zigbee network?
Is there any way to "restore" my Zigbee network?
If I can't, how do I clean house and start over again?

Most importantly:  If we're supposed to do something first, WHY does the software offer to upgrade if doing so destroys my Zigbee network?

**Comment by xyzroe** [comment, marked as answer]
1. Before performing the update, you should read the description of the update and understand the update process.
2. From your description, it is not entirely clear what is happening and what exactly is not working.
3. I recommend stopping z2m, clearing NVRAM via the Tools page under the Debug tab, rebooting the gateway, and starting z2m. If you see the devices in the web interface after starting z2m, then everything has gone well.
4. To start from scratch, you need to do everything described in step 3, but before starting z2m, delete the coordinator backup and database files located in the z2m folder. This will allow you to start the network from a clean slate.
5. **Most importantly:** Many users perform updates on a working system, and everything continues to work fine. All that is required is to stop before or restart z2m or zha just after update.
